be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -718,17 +717,22 @@ int64_t ExchangeSinkOperator::construct_brpc_attachment(const PTransmitChunkPara
         auto chunk = chunk_request->mutable_chunks(i);
         chunk->set_data_size(chunk->data().size());
 
-        int64_t before_bytes = CurrentThread::current().get_consumed_bytes();
-        attachment.append(chunk->data());
-        attachment_physical_bytes += CurrentThread::current().get_consumed_bytes() - before_bytes;
+        attachment_physical_bytes += chunk->data().size();
+        // release chunk->data() to iobuf
+        auto res = attachment.append_user_data((void*)((static_cast<std::string*>((chunk->release_data())))->c_str()),
+                                               chunk->data().size(), free);
```

Comment:
Could `free(c_str())` guarantee to release all the memory of `string`?

Reply:
other fields in string will deleted when deleting the shared ptr of chunk.

Reply:
As the document of [protobuf](https://developers.google.com/protocol-buffers/docs/reference/cpp-generated) says, the caller of `release_data()` takes the ownership of the allocated string object. 
Will the other fields in string really be deleted when the pchunk is destructed?

> string* release_foo(): Releases the ownership of the field and returns the pointer of the string object. After calling this, caller takes the ownership of the allocated string object, has_foo() will return false, and foo() will return the default value.

Reply:
changed, use ChunksDataRef to guarantee chunk's data is released after sent.